From 8335c8c20765d4f167d9b48e6a2757864a3bc7fd Mon Sep 17 00:00:00 2001
From: Eike Rathke <erack@redhat.com>
Date: Mon, 16 Nov 2020 23:26:56 +0100
Subject: Update to ICU 68.1

Also made it necessary to adapt two places in libcdr and libebook
that used UBool TRUE which is gone now to use standard true
instead.

Change-Id: I1c1df3030f8b883bec6045756907ee0b78060382
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/105964
Tested-by: Jenkins
Reviewed-by: Eike Rathke <erack@redhat.com>
---
 external/libcdr/UnpackedTarball_libcdr.mk     |  1 +
 external/libcdr/libcdr-no-icu-boolean.patch.1 | 12 ++++++++++++
 2 files changed, 13 insertions(+)
 create mode 100644 external/libcdr/libcdr-no-icu-boolean.patch.1

(limited to 'external/libcdr')

diff --git a/external/libcdr/UnpackedTarball_libcdr.mk b/external/libcdr/UnpackedTarball_libcdr.mk
index fd227a2f7445..778fd9eca123 100644
--- a/external/libcdr/UnpackedTarball_libcdr.mk
+++ b/external/libcdr/UnpackedTarball_libcdr.mk
@@ -17,6 +17,7 @@ $(eval $(call gb_UnpackedTarball_update_autoconf_configs,libcdr))
 
 $(eval $(call gb_UnpackedTarball_add_patches,libcdr, \
     external/libcdr/libcdr-visibility-win.patch \
+    external/libcdr/libcdr-no-icu-boolean.patch.1 \
 ))
 
 # vim: set noet sw=4 ts=4:
diff --git a/external/libcdr/libcdr-no-icu-boolean.patch.1 b/external/libcdr/libcdr-no-icu-boolean.patch.1
new file mode 100644
index 000000000000..65884fbbf873
--- /dev/null
+++ b/external/libcdr/libcdr-no-icu-boolean.patch.1
@@ -0,0 +1,12 @@
+diff -ur libcdr.org/src/lib/libcdr_utils.cpp libcdr/src/lib/libcdr_utils.cpp
+--- libcdr.org/src/lib/libcdr_utils.cpp	2020-02-02 16:30:44.000000000 +0100
++++ libcdr/src/lib/libcdr_utils.cpp	2020-11-16 21:08:04.092927628 +0100
+@@ -108,7 +108,7 @@
+     csd = ucsdet_open(&status);
+     if (U_FAILURE(status) || !csd)
+       return 0;
+-    ucsdet_enableInputFilter(csd, TRUE);
++    ucsdet_enableInputFilter(csd, true);
+     ucsdet_setText(csd, (const char *)buffer, bufferLength, &status);
+     if (U_FAILURE(status))
+       throw libcdr::EncodingException();
-- 
cgit v1.2.1
